app/src/main/java/com/oracle/javafx/scenebuilder/app/menubar/MenuBarController.java

```diff
@@ -1299,7 +1318,61 @@ private void updateZoomMenu() {
         }
     }
 
-    
+    private void runActionController(MenuItemController controllerToRun, KeyEvent event) {
+        if (controllerToRun.canPerform()) {
+            controllerToRun.perform();
+        }
+        event.consume();
+    }
+
+    /**
+     * Provides handling for additional keyboard accelerators assigned to document view
+     * zoom in/out operation. Zoom is controllable via [+]/[-] keys on standard keyboard
+     * and on numerical key pad.
+     * 
+     * Additionally zoom in via arrow keys (up/right) and zoom out (left/down) is possible.
+     * 
+     * @param event {@link KeyEvent} If any action is triggered, the event will be consumed.
+     */
+    public void handleAdditionalZoomAccelerators(KeyEvent event) {
+        // For unknown reasons, on macOS 12.0.1 with  Java 17/JavaFX 19 the "-" key code
+        // is not properly accepted hence this handling here.
+        if (EditorPlatform.IS_MAC && "-".equals(event.getText())) {
+            runActionController(zoomOutController, event);
+            return;
+        }
+
+        if (EditorPlatform.IS_MAC && "+".equals(event.getText())) {
+            runActionController(zoomInController, event);
+            return;
+        }
```

Comment:
There is also now special handling of `CMD`+`+` for MacOS in the MenuBar controller. Could you please try this @abhinayagarwal ?